src/sql/src/plan/statement/ddl/connection.rs

```diff
@@ -240,16 +240,28 @@ impl ConnectionOptionExtracted {
                         .availability_zones
                         .ok_or_else(|| sql_err!("AVAILABILITY ZONES option is required"))?,
                 };
-
                 if let Some(supported_azs) = scx.catalog.aws_privatelink_availability_zones() {
+                    let mut unique_azs: BTreeSet<String> = BTreeSet::new();
+                    let mut duplicate_azs: BTreeSet<String> = BTreeSet::new();
+                    // Validate each AZ is supported
                     for connection_az in &connection.availability_zones {
+                        if unique_azs.contains(connection_az) {
+                            duplicate_azs.insert(connection_az.to_string());
+                        } else {
+                            unique_azs.insert(connection_az.to_string());
+                        }
```

Comment:
slightly more succinct: 

```suggestion
                        let unique = unique_azs.insert(connection_az);
                        if !unique {
                            duplicate_azs.insert(connection_az.to_string());
                        }
```